orderbook/src/database/orders.rs

```diff
@@ -352,12 +358,30 @@ impl OrderStoring for Postgres {
                 (NOT invalidated) AND \
                 (NOT presignature_pending);"
         );
-        sqlx::query_as(QUERY)
-            .bind(min_valid_to as i64)
-            .fetch(&self.pool)
-            .err_into()
-            .and_then(|row: OrdersQueryRow| async move { row.into_order() })
-            .try_collect()
+        let mut connection = self.pool.acquire().await?;
+
+        connection
+            .transaction(move |transaction| {
+                async move {
+                    let orders = sqlx::query_as(QUERY)
+                        .bind(min_valid_to as i64)
+                        .fetch(&mut *transaction)
```

Comment:
I was struggling here because the compiler would say that in the next query below `transaction` had been moved in this line even though references shouldn't be moved and and `transaction` is a reference here. I don't fully understand the problem but it is related to `fetch` taking a generic parameter so the compiler treats the `transaction` as not being a reference. This is the fix I came up with.